src/TransitionSystems/transition_system.rs

```diff
@@ -17,6 +23,33 @@ pub enum PrecheckResult {
     NotConsistent(ConsistencyFailure),
 }
 
+#[derive(Clone, Copy)]
+/// Struct for determining the level for clock reduction
+pub struct Heights {
+    /// The level in the tree
+    pub(crate) tree: usize,
+    /// The level to reduce
+    pub(crate) target: usize,
+}
+
+impl Heights {
+    pub fn new(tree: usize, target: usize) -> Heights {
+        Heights { tree, target }
+    }
+
+    /// Function to "go down" a level in the tree
+    pub fn level_down(&self) -> Heights {
+        Heights {
+            tree: self.tree,
+            ..*self
+        }
+    }
```

Comment:
Should this function returns the same instance.
Your description of the function says it goes a level, but it still stays at the same level

Reply:
fixed